packages/space-opera/src/components/best_practices/render_best_practices.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {html} from 'lit-element';
+import {getModelViewer} from '../model_viewer_preview/reducer';
+
+export function renderProgressBar() {
+  getModelViewer()?.addEventListener('progress', onProgress);
+  return html`
+  <div class="progress-bar hide" slot="progress-bar">
+    <div class="update-bar"></div>
+  </div>
+  `
+}
+
+export const onProgress = (event) => {
```

Comment:
Is there a way we can reuse this for the best practices section or vice-versa? Maybe stringify the function?

Reply:
We can't directly stringify a function, and using toString() on it, will convert it to the undreadable js.

Reply:
Do you already have a dependency on js-beautify? You could try this: https://stackoverflow.com/a/39544613/7852087